terraform-tests/s3_test.go

```diff
@@ -116,4 +117,23 @@ var _ = Describe("S3", Label("S3-terraform"), Ordered, func() {
 				}))
 		})
 	})
+
+	Context("explicitly denies access to HTTP requests", func() {
+		BeforeAll(func() {
+			plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{"restrict_to_tls_requests_only": true}))
+		})
+
+		It("should create the right resources", func() {
```

Comment:
```suggestion
		It("should create an aws_s3_bucket_policy", func() {
```

Reply:
It would be great to be able to check not only the type of resource but also its name in Terraform tests.
I don't know if it's possible or not, and its not really that important for this test.

Reply:
It's not possible. The values are known after the apply phase. I added the suggestions! Thank you very much!

Reply:
I was referring to the name of the resource in the HCL specification (`bucket_require_tls_policy` from the code below):

https://github.com/cloudfoundry/csb-brokerpak-aws/blob/98267b411e09b590c3d99f90d0727b2cb1db343c/terraform/s3/provision/main.tf#L93

Currently it doesn't provide too much value since there aren't any other resources with type `aws_s3_bucket_policy`.

It seems it is possible to check the name and type of a resource in our tests:

I will create a new helper function to do exactly that in the future. For this test it is perfect as it is :)
Thanks!